WooCommerce/WooCommerceTests/POS/Mocks/MockPOSOrderService.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+@testable import Yosemite
+
+class MockPOSOrderService: POSOrderServiceProtocol {
+    var orderToReturn: Order?
+    func syncOrder(cart: [Yosemite.POSCartItem], order: Yosemite.Order?, allProducts: [any Yosemite.POSItem]) async throws -> Yosemite.Order {
```

Comment:
Unrelated to this PR, it just came up because the return type of this mock and the unit tests.

I see we're using `OrderItem` and `Order` in several places within POS scope (`MockPOSOrderService`, `POSOrderPreviewService`, etc, ... ), rather than the previous POS versions (eg: `POSOrder`). The change was done [here](https://github.com/woocommerce/woocommerce-ios/pull/13381) when addressing tax calculation, but this change seems to break the differentiation between `app` and `pos`?

I have no specific considerations at the moment, just raising this up in case it is something we want to re-evaluate. I left a comment in that PR for the time being.

Reply:
Yeah, I would prefer if we could nip this in the bud as well, and go back to using simpler objects in the POS code.